Coding/fw/data/ws.html

```diff
@@ -0,0 +1,468 @@
+<!--
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+-->
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>ESP Web Server</title>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <link rel="stylesheet" href="/css/style.css" />
+    <script src="/js/ws.js"></script>
+    <script>
+        var logMessages = [];
+        var maxLogs = 40;
+        var counter = 1;
+        var enable = 1;
+
+        var gateway = `ws://${window.location.hostname}/ws`;
+        var websocket;
+        window.addEventListener('load', onLoad);
+        window.addEventListener("keyup", function(event) {
+            if (event.keyCode === 13) {
+                event.preventDefault();
+                document.getElementById("sendCommandButton").click();
+                document.getElementById("sendFrameButton").click();
+            }
+        });
+
+        function initWebSocket() {
+            console.log('Trying to open a WebSocket connection...');
+            websocket = new WebSocket(gateway);
+            websocket.onopen = onOpen;
+            websocket.onclose = onClose;
+            websocket.onmessage = onMessage;
+        }
+
+        function onOpen(event) {
+            console.log('Connection opened');
+            websocket.send("D\r");
+        }
+
+        function onClose(event) {
+            console.log('Connection closed');
+            setTimeout(initWebSocket, 2000);
+        }
+
+        function onLoad(event) {
+            initWebSocket();
+            setInterval(getCMD, 10);
+        }
+
+        function onMessage(event) {
+            logMessages.push(event.data);
+        }
+
+        function getCMD() {
+            if (logMessages.length != 0) {
+                if (enable == 1) {
+
+                    console.log(logMessages[0]);
+
+                    switch (logMessages[0][0]) {
+                        case "\r":
+                            console.log("OK");
+                            break;
+                        case ".":
+                            console.log("Error");
+                            break;
+
+                        case "t":
+                        case "T":
+                        case "r":
+                        case "R":
+                            displayMessages();
+                            break;
+
+                        case "D":
+                            displaySettings();
+                            break;
+
+                        default:
+                            console.log("Unrecognized Lawicel Command");
+                            break;
+                    }
+                    logMessages.shift();
+                }
+            }
+        }
+
+        function displayMessages() {
```

Comment:
shoudn't the message to be displayed pushed as parameter instead of accessing the global array ?